egs2/wsj0-2mix/enh1/enh.sh

```diff
@@ -0,0 +1,682 @@
+
+    key_file="${_enh_train_dir}/${_scp}"
+    split_scps=""
+    for n in $(seq "${_nj}"); do
+        split_scps+=" ${_logdir}/train.${n}.scp"
+    done
+    # shellcheck disable=SC2086
+    utils/split_scp.pl "${key_file}" ${split_scps}
+
+    key_file="${_enh_dev_dir}/${_scp}"
+    split_scps=""
+    for n in $(seq "${_nj}"); do
+        split_scps+=" ${_logdir}/dev.${n}.scp"
+    done
+    # shellcheck disable=SC2086
+    utils/split_scp.pl "${key_file}" ${split_scps}
+
+    # 2. Submit jobs
+    log "Enhancement collect-stats started... log: '${_logdir}/stats.*.log'"        
+
+    # prepare train and valid data parameters
+    _train_data_param="--train_data_path_and_name_and_type ${_enh_train_dir}/wav.scp,speech_mix,sound "
+    _valid_data_param="--valid_data_path_and_name_and_type ${_enh_dev_dir}/wav.scp,speech_mix,sound "
+    for spk in $(seq "${spk_num}"); do
+        _train_data_param+="--train_data_path_and_name_and_type ${_enh_train_dir}/spk${spk}.scp,speech_ref${spk},sound "
+        _valid_data_param+="--valid_data_path_and_name_and_type ${_enh_dev_dir}/spk${spk}.scp,speech_ref${spk},sound "
+    done
+
+    if $use_dereverb_ref; then
+        # reference for dereverberation
+        _train_data_param+="--train_data_path_and_name_and_type ${_enh_train_dir}/dereverb.scp,dereverb_ref,sound "
+        _valid_data_param+="--valid_data_path_and_name_and_type ${_enh_dev_dir}/dereverb.scp,dereverb_ref,sound "
+    fi
+
+    if $use_noise_ref; then
+        # reference for denoising
+        _train_data_param+=$(for n in $(seq $noise_type_num); do echo -n \
+            "--train_data_path_and_name_and_type ${_enh_train_dir}/noise${n}.scp,noise_ref${n},sound "; done)
+        _valid_data_param+=$(for n in $(seq $noise_type_num); do echo -n \
+            "--valid_data_path_and_name_and_type ${_enh_dev_dir}/noise${n}.scp,noise_ref${n},sound "; done)
+    fi
+
+    # NOTE: --*_shape_file doesn't require length information if --batch_type=unsorted,
+    #       but it's used only for deciding the sample ids.
+
+
+    # shellcheck disable=SC2086
+    ${train_cmd} JOB=1:"${_nj}" "${_logdir}"/stats.JOB.log \
+        python3 -m espnet2.bin.enh_train \
+            --collect_stats true \
+            --use_preprocessor true \
+            ${_train_data_param} \
+            ${_valid_data_param} \
+            --train_shape_file "${_logdir}/train.JOB.scp" \
+            --valid_shape_file "${_logdir}/dev.JOB.scp" \
+            --output_dir "${_logdir}/stats.JOB" \
+            ${_opts} ${enh_args} \
+            --batch_type unsorted
+
+    # 3. Aggregate shape files
+    _opts=
+    for i in $(seq "${_nj}"); do
+        _opts+="--input_dir ${_logdir}/stats.${i} "
+    done
+    # shellcheck disable=SC2086
+    python3 -m espnet2.bin.aggregate_stats_dirs ${_opts} --output_dir "${enh_stats_dir}"
+
+    # Append the num-tokens at the last dimensions. This is used for batch-bins count
```

Comment:
remove these comment parts.
We do not need text related processing.

Reply:
Done.